core/src/main/java/ai/timefold/solver/core/api/solver/SolverJobBuilder.java

```diff
@@ -77,16 +79,30 @@ public interface SolverJobBuilder<Solution_, ProblemId_> {
             withFinalBestSolutionConsumer(@NonNull Consumer<? super Solution_> finalBestSolutionConsumer);
 
     /**
-     * Sets the consumer of the first initialized solution. First initialized solution is the solution at the end of
-     * the last phase that immediately precedes the first local search phase. This solution marks the beginning of actual
-     * optimization process.
+     * As defined by #withFirstInitializedSolutionConsumer(FirstInitializedSolutionConsumer).
+     *
+     * @deprecated Use {@link #withFirstInitializedSolutionConsumer(FirstInitializedSolutionConsumer)} instead.
+     */
+    @Deprecated(forRemoval = true, since = "1.19.0")
+    @NonNull
+    default SolverJobBuilder<Solution_, ProblemId_>
+            withFirstInitializedSolutionConsumer(@NonNull Consumer<? super Solution_> firstInitializedSolutionConsumer) {
+        return withFirstInitializedSolutionConsumer(
+                (solution, isTerminatedEarly) -> firstInitializedSolutionConsumer.accept(solution));
+    }
+
+    /**
+     * Sets the consumer of the first initialized solution,
+     * the beginning of the actual optimization process.
+     * First initialized solution is the solution at the end of the last phase
+     * that immediately precedes the first local search phase.
      *
      * @param firstInitializedSolutionConsumer called only once before starting the first Local Search phase
      * @return this
      */
     @NonNull
-    SolverJobBuilder<Solution_, ProblemId_>
-            withFirstInitializedSolutionConsumer(@NonNull Consumer<? super Solution_> firstInitializedSolutionConsumer);
+    SolverJobBuilder<Solution_, ProblemId_> withFirstInitializedSolutionConsumer(
+            @NonNull FirstInitializedSolutionConsumer<? super Solution_> firstInitializedSolutionConsumer);
```

Comment:
@cristianonicolai You asked me to notify you of the change. This is it - the consumer gets an extra argument as discussed, the old method gets deprecated but still works.